@here/olp-sdk-dataservice-read/lib/utils/getTile.ts

```diff
@@ -61,109 +117,187 @@ import {
+    request: TileRequest,
     abortSignal?: AbortSignal
+): Promise<Response>;
+
+export async function getTile(
+    request: TileRequest,
+    paramsOrSignal?: TileRequestParams | AbortSignal,
+    signal?: AbortSignal
 ): Promise<Response> {
-    const layerType = rq.getParams().layerType;
-    const settings = rq.getParams().settings;
-    const catalogHrn = rq.getParams().catalogHrn;
-    const layerId = rq.getParams().layerId;
-    const billingTag = rq.getBillingTag();
-    const quadKey = rq.getTileKey();
-    const fetchOptions = rq.getFetchOption();
-    const cache = new MetadataCacheRepository(settings.cache);
+    let params: TileRequestParams | undefined;
+    let abortSignal: AbortSignal | undefined;
 
+    if (
+        !paramsOrSignal ||
+        (paramsOrSignal && paramsOrSignal instanceof AbortSignal)
+    ) {
+        // usind deprecated
+        params = request.getParams();
+        abortSignal = paramsOrSignal;
+    } else {
+        params = paramsOrSignal;
+        abortSignal = signal;
+    }
+
+    const quadKey = request.getTileKey();
     if (!quadKey) {
         return Promise.reject(new Error("Please provide correct QuadKey"));
     }
 
-    const partitionId = `${mortonCodeFromQuadKey(quadKey)}`;
-    const partitionsRequest = new PartitionsRequest().withPartitionIds([
-        partitionId
-    ]);
+    if (!params) {
+        return Promise.reject(
+            new Error(
+                `Error getting params. Use TileRequest.withParams() method.`
+            )
+        );
+    }
+
+    let catalogVersion: number | undefined;
+    let blobType: ApiName = "volatile-blob";
+
+    if (params.layerType === "versioned") {
+        blobType = "blob";
+        catalogVersion = await request.getCatalogVersion();
+    }
 
-    const blobType = layerType === "versioned" ? "blob" : "volatile-blob";
-    const queryClient = new QueryClient(settings);
     const blobRequestBuilder = await RequestFactory.create(
         blobType,
         "v1",
-        settings,
-        catalogHrn,
+        params.settings,
+        params.catalogHrn,
         abortSignal
     ).catch(error => Promise.reject(error));
 
-    if (fetchOptions !== FetchOptions.OnlineOnly) {
-        const cachedPartition = cache.get(
-            partitionsRequest,
-            catalogHrn.toString(),
-            layerId
+    const delta = 4;
+    const requestedTileKey = TileKey.fromRowColumnLevel(
+        quadKey.row,
+        quadKey.column,
+        quadKey.level
+    );
+    const parentTileKey = requestedTileKey.changedLevelBy(-delta);
+
+    let quadTreeIndex = null;
+
+    if (request.getFetchOption() !== FetchOptions.OnlineOnly) {
+        const cache = new QuadTreeIndexCacheRepository(params.settings.cache);
+
+        const cachedTree = cache.get({
+            hrn: params.catalogHrn.toString(),
+            layerId: params.layerId,
+            depth: delta,
+            root: parentTileKey,
+            version: catalogVersion
+        });
```

Comment:
This is a tricky call. You should not always request with exactly tilekey - 4 level as this will always get you duplicate indexes. E.g. users first tile was level 14, means quad tree root is level 10; then user requests tile level 13, means quad tree root is level 9, means you dont find it in cache. But the level 13 tile is actually in the level 10 quad tree requested before.
The algorithm is actually to scan the cache for each level up to -4.
So for a level 13 tile you check level 12, level 11, level 10 and level 9 in cache, only then you go and request online.

Reply:
added that, thanks